occurrence-download/src/test/java/org/gbif/occurrence/download/file/archive/MultiDirectoryArchiveBuilderTest.java

```diff
@@ -0,0 +1,95 @@
+package org.gbif.occurrence.download.file.archive;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.LocalFileSystem;
+import org.junit.jupiter.api.Test;
+
+import org.gbif.hadoop.compress.d2.D2Utils;
+import org.gbif.hadoop.compress.d2.zip.ModalZipOutputStream;
+import org.gbif.utils.file.FileUtils;
+import org.gbif.utils.file.InputStreamUtils;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.net.URI;
+import java.nio.file.Files;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class MultiDirectoryArchiveBuilderTest {
+
+  @Test
+  public void testBuildDefaultMode() throws Exception {
+    FileSystem sourceFileSystem = new LocalFileSystem();
+    sourceFileSystem.initialize(URI.create("file:///"), new Configuration());
+
+    String[] arguments = {
+      FileUtils.getClasspathFile("multitsv/default/first").getAbsolutePath(),
+      "first.csv",
+      "col1,col2,col3",
+
+      FileUtils.getClasspathFile("multitsv/default/second").getAbsolutePath(),
+      "second.csv",
+      "",
+
+      FileUtils.getClasspathFile("multitsv/default/third").getAbsolutePath(),
+      "third.csv",
+      "colⅠ,colⅡ,colⅢ",
+
+      FileUtils.createTempDir().getAbsolutePath(),
+      "empty.csv",
+      "col一,col二,col三"
+    };
+    String targetPath = Files.createTempDirectory("multidir-default").toString();
+    String downloadKey = "testArchive";
+
+    MultiDirectoryArchiveBuilder.withEntries(arguments)
+      .mergeAllToZip(sourceFileSystem, sourceFileSystem, targetPath, downloadKey,
+        ModalZipOutputStream.MODE.DEFAULT);
+
+    ZipFile zf = new ZipFile(targetPath + "/testArchive.zip");
+    ZipEntry ze = zf.getEntry("third.csv/000003");
+    assertEquals("η,θ,ι\n", new InputStreamUtils().readEntireStream(zf.getInputStream(ze)));
+  }
+
+  @Test
+  public void testBuildPreDeflatedMode() throws Exception {
```

Comment:
this test case failed: 
java.lang.IllegalArgumentException: invalid entry crc-32

	at org.gbif.hadoop.compress.d2.zip.ZipEntry.setCrc(ZipEntry.java:179)
	at org.gbif.occurrence.download.file.archive.MultiFileArchiveBuilder.zipPreDeflated(MultiFileArchiveBuilder.java:150)
 it seems that the ORC > 0xFFFFFFFFL
see: [https://github.com/gbif/hadoop-compress/blob/master/src/main/java/org/gbif/hadoop/compress/d2/zip/ZipEntry.java#L178](https://github.com/gbif/hadoop-compress/blob/master/src/main/java/org/gbif/hadoop/compress/d2/zip/ZipEntry.java#L178)